tutorials/working_with_auxiliary_data.md

```diff
@@ -0,0 +1,88 @@
+
+For convenience, assume that we are fitting and evaluating a recommender model leveraging item textual auxiliary data.  As a running example, let's consider the Collaborative Deep Learning ([CDL](../cornac/models/cdl)) model, which is already implemented in Cornac.
+ 
+ 
+### Dataset
+We use the well know MovieLens 100K dataset. It consists of user-movie interactions in the triplet format `(user_id, movie_id, rating)`, as well as movie plots in the format `(movie_id, text)`, which represent our textual auxiliary information. This dataset is already accessible from Cornac, and we can load it as follows,
+```Python
+import cornac
+from cornac.data import Reader
+from cornac.datasets import movielens
+
+plots, movie_ids = movielens.load_plot()
+rating_data = movielens.load_100k(reader=Reader(item_set=movie_ids, bin_threshold=3))
+```
+where we have filtered out movies without plots and binarized the integer ratings using `cornac.data.Reader`.
+
+### The TextModality class
+
+With our dataset in place, the next step is to instantiate a `TextModality`, which will allow us to manipulate and represent our auxiliary data in the desired format.  
+ ```Python
+from cornac.data import TextModality
+from cornac.data.text import BaseTokenizer
+
+item_text_modality = TextModality(corpus=plots, ids=movie_ids, 
+                                  tokenizer=BaseTokenizer(sep='\t', stop_words='english'),
+                                  max_vocab=5000, max_doc_freq=0.5)
+```
+In addition to the movie plots and ids, we have specified a `cornac.data.text.Tokenizer` to split text, we limited the maximum vocabulary size to 5000, as well as filtered out words occurring in more than 50% of documents (plots in our case) by setting `max_doc_freq = 0.5`. For more options/details on the `TextModality` please refer to the [docs](https://cornac.readthedocs.io/en/latest/data.html#module-cornac.data.text). 
+ 
+ 
+**Bag-of-Words text representation.** CDL assumes the bag-of-words representation for text information, i.e., in the form of a document-word matrix. The good news is that we don't have to worry about how to generate such representation from our raw texts. The `TextModality` class implements the necessary routines to process and output different representations for text data, e.g., sequence, bag of words, tf-idf. That is, to get our auxiliary data under the desired format, all we need inside the `CLD` [implementation](../cornac/models/cdl/recom_cld.py) is the following line of code:
+```Python
+text_feature = self.train_set.item_text.batch_bow(np.arange(n_items))
+``` 
+where `self.train_set.item_text` is our `item_text_modality`, `n_items` is number of training items, and the `batch_bow()` function returns the bag-of-words vectors of the specified item ids, in our case we want the text features for all training items. In more details, the rows of `text_feature` correspond to the bag-of-words vectors of the provided item ids to `batch_bow()`.
```

Comment:
where `self.train_set.item_text` is our `item_text_modality`, `n_items` is `the` number of training items, and the `batch_bow()` function returns the bag-of-words vectors of the specified item ~ids~`indices`, in our case we want the text features for all training items. In more details, the rows of `text_feature` correspond to the bag-of-words vectors of the provided item ~ids~`indices` to `batch_bow()`.